Source/Mothership/MothershipDummy.cpp

```diff
@@ -0,0 +1,846 @@
+#include "MothershipDummy.h"
```

Comment:
MothershipDummy is, as you say, not interacting with the HostLink, but in addition, it's configured to respond to OnReply messages (which are responses to NameServer requests). 

This is particularly useful for testing because it means you don't need a querying external process (which would have to be handled through the nonexistent UserIO process), you can just issue queries from MothershipDummy and then get responses back from the NameServer. The ConfigState command (Q::NAME,Q::CFG,Q::STATE) in the message key) triggers MothershipDummy to run some queries. 

Meanwhile, a regular Mothership should definitely not respond like MothershipDummy does to OnReply messages - it should just invoke the SBase class' default handler for that (warn and continue), and it MUST not respond to ConfigState messages (because the  state of a Mothership explicitly depends on actions that interact with the HostLink - thus a command from outside to change its state should do nothing), so deriving from MothershipDummy is not appropriate. 

You are correct that it's there mostly for testing - to emulate within reason a 'Mothership' without actually being a HostLink as well. I suspect though that when we come to test UserIO it will be valuable in any case. MothershipDummy also shows how a lot of the system was tested - by deploying tasks to it and having it output to a file the results it received after running a series of queries (we have to test the query functionality!)